cloudigrade/config/settings/prod.py

```diff
@@ -2,11 +2,24 @@
 import sentry_sdk
 from sentry_sdk.integrations.django import DjangoIntegration
 
+from util.aws.sts import _get_primary_account_id
 from .base import *
 
 IS_PRODUCTION = CLOUDIGRADE_ENVIRONMENT == "prod"
 
-DEBUG = env.bool("DJANGO_DEBUG", default=False)
+DJANGO_DEBUG = "False"
+
+CLOUDIGRADE_ENVIRONMENT = env("CLOUDIGRADE_ENVIRONMENT")
+
+HOUNDIGRADE_AWS_AVAILABILITY_ZONE = "us-east-1a"
+HOUNDIGRADE_AWS_AUTOSCALING_GROUP_NAME = f"cloudigrade-ecs-asg-{CLOUDIGRADE_ENVIRONMENT}"
+HOUNDIGRADE_ECS_CLUSTER_NAME = f"cloudigrade-ecs-{CLOUDIGRADE_ENVIRONMENT}"
+
+AWS_NAME_PREFIX = f"{CLOUDIGRADE_ENVIRONMENT}-"
+
+AWS_CLOUDTRAIL_EVENT_URL = f"https://sqs.us-east-1.amazonaws.com/{_get_primary_account_id()}/cloudigrade-cloudtrail-s3-{CLOUDIGRADE_ENVIRONMENT}"
```

Comment:
Calling our code _during_ config loading feels fragile and risky to me. I checked the sequence of imports, and it _should_ be okay for now, but if we aren't very careful around the `sts` and `arn` packages, we could unknowingly introduce something like an import loop that we wouldn't see until the new code fails to start in "prod" (yes this config is also used for stage, etc). If we introduce something later in the chain needs to use something from `from django.conf import settings`, I suspect things could go badly here.


I pondered at https://github.com/cloudigrade/cloudigrade/pull/814/files#r614997276 if we could ask `aws_caller_info` for the AWS account ID. Should we consider doing something like that and just explicitly set an env var with the AWS account ID? On one hand, that would generally be redundant with the keys. On the other hand, "explicit is better than implicit"?

Any other thoughts or possible alternative solutions?

Reply:
Well we can't use Ansible to do it, because there is no way to get data from ansible to the deployment. Only vars we explicitly bake in, or set in the config map. "Setting" an account number to a set value once again drives us further away from being able to just get aws keys and run in any environment.

I dislike this strongly, but I see no other way to continue as I can't just provide a templated .env file that would be ready by all the pods.